### DISCUSSION TITLE
Late to the ResteasyServerCommonProcessor.build party

### DISCUSSION THREAD

**Discussion by ivlcic** [quarkusio/quarkus#39694] [Q&A] [answered]
Hi, all

I'm trying to create an extension, a build step, that replaces or adds the `jakarta.ws.rs.Path` annotation on my custom target classes while using Resteasy Classic.
As it seems, the best thing would be to introduce a build step in my extension deployment with a `BuildProducer<AnnotationsTransformerBuildItem>` parameter that produces such transformation.

For example:
```java
  @BuildStep
  public void processPathAnnotation(
      ...
      BuildProducer<AnnotationsTransformerBuildItem> transformerProducer) {
      ...
      transformerProducer.produce(
          new AnnotationsTransformerBuildItem(
              new AnnotationsTransformer.ClassTransformerBuilder()
                  .whenClass(o -> o.name().equals(mapping.implClass.name()))
                  .thenTransform(
                      transformation -> transformation.add(
                          PATH_ANNOTATION_DOT_NAME,
                          AnnotationValue.createStringValue("value", newPath)
                      )
                  )
          )
      );
      ...
```
But my transformation is always invoked after `io.quarkus.resteasy.server.common.deployment.ResteasyServerCommonProcessor#build`, and I don't know what the best way to achieve this would be, maybe which build items to consume to transform earlier in the build process.

I would appreciate any help!

p.s. I've tried many things (`BytecodeTransformerBuildItem` manipulation `ResteasyDeploymentCustomizerBuildItem` ..., but I'm always late to the party or just too novice)

best
Nikola

**Comment by gsmet** [comment]
Hi!

`AnnotationsTransformerBuildItem` won't work because it's a layer on top of the Jandex index that is only used by ArC, our CDI implementation.

Same for bytecode transformation unfortunately because the bytecode transformation is executed after the classes are indexed in Jandex. And RESTEasy Classic uses the Jandex index.

Now maybe you should have a look at RESTEasy Reactive (that we will rename Quarkus REST in 3.9, can be used for both reactive and blocking and is our default REST layer now) because I know it has a layer similar to annotations transformer (but specific to Quarkus REST). @geoand will be able to tell you if what you want to do is doable.

Also it might be a good idea to have a look at the Quarkus Spring Web extensions that provide a compatibility layer for Spring Web and actually registers REST resources (there is an extension for Quarkus REST and another for RESTEasy Classic, with a few things shared between the two).

**Reply by geoand** [reply, marked as answer]
I completely agree with everything @gsmet said.

What you are trying to do will be fairly easy with the custom annotation transformer API that Quarkus REST provides

**Reply by ivlcic** [reply]
I'll take a deeper look at the reactive code. SpringWeb style of integration with registering MyAdditionalAnno is a kind of catch-22 for me: I can't retrofit classes with my-custom-registering-annotation (if I could do that, I'd rather use Jaxrs's Path than my custom stuff). Additionally, I'd love to provide OpenAPI web-ui, which Quarkus generates. 
My aim is to generate REST paths and register resources that play nicely and by the book with Quarkus's rules so that I won't spoil other contributors' stuff.
I've got my eyes on' NonJaxRsClassBuildItem`, which would be a super-cool feature to use, but as I understand it, it is only intended to be used for Restasy Standalone stuff (probably for testing only).

Thanks, guys, for the answers and effort - I do admire your work.

**Reply by geoand** [reply]
Let us know if you need any help :)
